anomalib/data/folder.py

```diff
@@ -139,6 +139,13 @@ def make_dataset(
             if row.label_index == 1:
                 samples.loc[index, "mask_path"] = str(mask_dir / row.image_path.name)
 
+    # make sure all the files exist, dirs_str is used for error message
+    dirs_str = f"normal_dir={normal_dir}, abnormal_dir {abnormal_dir}, normal_test_dir={normal_test_dir}"
+    assert samples.image_path.apply(lambda x: x.exists()).all(), f"missing image files, {dirs_str}"
```

Comment:
Do we need this check? Isn't `image_path` populated by all the images in the directory. We populate mask_path based on the image in the image directory so the second check would make sense.

Reply:
good point